### PR TITLE
Store user object in session

### LINKED ISSUE #25: Store user in session [1]
The user object should be stored in the session

### PR DESCRIPTION
**Pull request by subnomo**
Fixes #25 
## Changes in this PR.

- [X] Remove username from session object
- [X] Add user object to session object
## Testing this PR.

1. Run `nosetests`
### Expected Output.

```
$ nosetests
.......................
----------------------------------------------------------------------
Ran 23 tests in 0.243s

OK
```

@osuosl/devs

### CODE REVIEW

File: app/views.py [hunk 1]
```diff
@@ -30,14 +30,14 @@ def isLoggedIn():
...
     if not isLoggedIn():
         return {'error': "Not logged in."}
 
     ts = pymesync.TimeSync(baseurl=app.config['TIMESYNC_URL'],
                            test=app.config['TESTING'],
                            token=session['token'])
-    user = ts.get_users(username=session['username'])
+    user = ts.get_users(username=username)
 
     return user
```
Comment (aaroncohen73) [inline app/views.py]: I think that you should check for errors before returning the user object. Otherwise, we might get some unexpected errors when other parts of the code try to access the user object.

Comment (subnomo) [inline app/views.py]: @aaroncohen73 Ok. What should the method return if there are errors?

Comment (aaroncohen73) [inline app/views.py]: @subnomo You should return an empty dictionary so that other parts of the code can just go

```
if not user:
    # ...
```

to check if the user object gotten from get_user() is valid

Comment (subnomo) [inline app/views.py]: @aaroncohen73 What if we need to flash the error message to the user? By returning an empty dict we have no idea what went wrong.

File: app/views.py [hunk 2]
```diff
@@ -87,8 +87,19 @@ def login():
             return "There was an error.", 500
         # Else success, redirect to index page
         else:
-            session['username'] = username
             session['token'] = token['token']
+            user = getUser(username)
+
+            if type(user) is list:
```
Comment (pop) [inline app/views.py]: The `getUser` method should only return a string. We should fix that now.

**Comment by pop** [comment]
@subnomo fix the conflicts in this branch when you get a chance. I have a little bit of feedback, but pending those being addressed and the merge conflicts being resolved I give this a +1.

File: app/views.py (hunk 1 above)
Comment (aaroncohen73) [inline app/views.py]: @subnomo We don't need to flash a detailed, technical message to the user. We can just flash a general error message to the user and then print the error to the console or a logfile inside get_user() before the empty dict is returned. That's what you're doing already at line 99.

File: app/views.py (hunk 2 above)
Comment (subnomo) [inline app/views.py]: @ElijahCaine It should return the user object, so we can store it in the session.

And until the changes made in osuosl/pymesync#140 are pushed to PyPI we still have to check if the user is a list.

**Comment by aaroncohen73** [comment]
+1 pending rebase

File: app/views.py (hunk 2 above)
Comment (pop) [inline app/views.py]: Let me rephrase that: the `getUser` method (singular) (`get_user` for consistent naming?) should not return _multiple_ user**s**.

EDIT: The list data type implies that the function might return multiple of anything. The function `getUser` should only ever return a singular thing so it should return that thing and not a list.

Comment (aaroncohen73) [inline app/views.py]: @ElijahCaine When supplied with a username argument, TimeSync.get_users() will return at most one user.

Comment (pop) [inline app/views.py]: @aaroncohen73 I get that. Which is why this check should not exist. The function we are calling to set the `user` variable should just return the user and not a list containing the user.
What we have now:

``` python
user = getUser(username)

if type(user) is list:    # A silly check to make because that function shouldn't return a list data type.
    user = user[0]
```

What we should have

``` python
user = getUser(username)    #only returns user object and not list because why would it?
```

Comment (aaroncohen73) [inline app/views.py]: @ElijahCaine @subnomo Then this is another check that should be put inside `get_user()`?

Comment (subnomo) [inline app/views.py]: @ElijahCaine Ah okay, I see what you mean.

Comment (subnomo) [inline app/views.py]: @aaroncohen73 A single check, so we don't have to check later.

Comment (pop) [inline app/views.py]: @aaroncohen73 Correct. If `TimeSync.get_users()` ever returns a list that check should be done in `get_user(username=...)` and only the user object should be returned. If it only ever returns the user no checks need to be made.

**Comment by pop** [comment]
@subnomo re f3f7f96,

excellent

+1

**Comment by subnomo** [comment]
Ugh, gotta rebase again.

**Comment by Kennric** [comment]
Because of the way views have been moved around, I think it is going to be very difficult to rebase - it may be better to tackle this as a new issue using the latest develop and modify the moved views

**Comment by Kennric** [comment]
superseded by PR #71
